src/test_l3_plat_power.c

```diff
@@ -0,0 +1,346 @@
+   UT_LOG_MENU_INFO("Supported Wake Up Source Types enabled/disabled");
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("\t#  %-20s","Enable/Disable Wakeup Source");
+   UT_LOG_MENU_INFO("\t%d  %-20s", 0, "Disable");
+   UT_LOG_MENU_INFO("\t%d  %-20s", 1, "Enable");
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("Select Enable/Disable Wakeup Source: ");
+   scanf("%d", &enableGet);
+   readAndDiscardRestOfLine(stdin);
+   if(sourceType != 0 && sourceType != 1)
+        {
+            UT_LOG_ERROR("Invalid enable type");
+            goto exit;
+        }
+    bool enable = (bool)enableGet;
+   // Step 1: Call PLAT_DS_INIT()
+   status = PLAT_API_SetWakeupSrc( srcType, enable );
+   UT_LOG_INFO("Result PLAT_API_SetWakeupSrc: pmStatus_t:[%s]",
+                UT_Control_GetMapString(pmStatus_mapTable, status));
+
+   exit:
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief Initialization of the HAL Power Manager Module
+*
+* This test provides a scope to get what wake up sources are active.
+
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 003@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_power_manager_hal_Get_Wakeup_Source(void)
+{
+   gTestID = 2;
+   pmStatus_t status = PWRMGR_SUCCESS ;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   int32_t  sourceType =0;
+   bool  enable = false;
+
+
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("Supported Wake Up Source Types");
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("\t#  %-20s","Wakeup Source Type");
+   for(int32_t i = PWRMGR_WAKEUPSRC_VOICE; i <= PWRMGR_WAKEUPSRC_LAN; i++)
+   {
+       UT_LOG_MENU_INFO("\t%d.  %-20s", i, UT_Control_GetMapString(plat_source_types_mapTable, i));
+   }
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("Select Wake Up Source Type: ");
+   scanf("%d", &sourceType);
+   readAndDiscardRestOfLine(stdin);
+   if(sourceType < PWRMGR_WAKEUPSRC_VOICE || sourceType > PWRMGR_WAKEUPSRC_LAN)
+        {
+            UT_LOG_ERROR("Invalid ARC Type");
+            goto exit;
+        }
+   PWRMGR_WakeupSrcType_t srcType = (PWRMGR_WakeupSrcType_t)sourceType;
+   // Step 1: Call PLAT_DS_INIT()
+   status = PLAT_API_GetWakeupSrc(srcType,*enable);
+   if ((status != PWRMGR_SUCCESS ))
+   {
+       UT_LOG_INFO("Result PLAT_API_GetWakeupSrc: pmStatus_t:[%s]",
+                UT_Control_GetMapString(pmStatus_mapTable, status));
+   }
+   else{
+    UT_LOG_MENU_INFO("\t%d.  %-20s is ", sourceType, UT_Control_GetMapString(plat_source_types_mapTable, sourceType), enable ? "enabled" : "disabled");
```

Comment:
Here in the log print statement boolean print is missing only 2 values are added. Please check and update.

Reply:
Updated as requested.